### PR TITLE
fix nodeSelector invalid and reconcile conflict err

### PR DESCRIPTION
**Pull request by lgy1027**
Fixed using master helm install, nodeSelector specified by Exporter does not work

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@junotx Please help to review this

File: pkg/controllers/exporter_controller.go [hunk 1]
```diff
@@ -125,26 +125,43 @@ func (r *ExporterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
+			log.V(1).Info("Conflict while updating status", "namespace", cm.Namespace, "cm_name", cm.Name)
+			return reconcile.Result{Requeue: true}, nil
+		}
 		return ctrl.Result{}, e
 	}
 	deploy := &appsv1.Deployment{}
 	deploy.Name = kee.Name
 	deploy.Namespace = kee.Namespace
 	if _, e = controllerutil.CreateOrUpdate(ctx, r.Client, deploy, r.deployMutate(deploy, cm, sa, kee)); e != nil {
+		log.V(1).Info("deploy nodeSelect", "nodeSelect", deploy.Spec.Template.Spec.NodeSelector)
```
Comment (junotx) [inline pkg/controllers/exporter_controller.go:157]: This log may be set a higher verbosity level to mark it less important @lgy1027

File: pkg/controllers/exporter_controller.go [hunk 2]
```diff
@@ -125,26 +125,43 @@ func (r *ExporterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
...
+		if apierrs.IsConflict(e) {
+			log.V(1).Info("Conflict while updating status", "namespace", deploy.Namespace, "deploy_name", deploy.Name)
+			return reconcile.Result{Requeue: true}, nil
+		}
 		return ctrl.Result{}, e
 	}
-
+	log.V(1).Info("deploy nodeSelect", "nodeSelect", deploy.Spec.Template.Spec.NodeSelector)
```
Comment (junotx) [inline pkg/controllers/exporter_controller.go:164]: Same as above, this log may be set a higher verbosity level to mark it less important

**Review by junotx — CHANGES_REQUESTED** [review]

File: pkg/controllers/exporter_controller.go (hunk 2 above)
Comment (lgy1027) [reply pkg/controllers/exporter_controller.go:164]: > Same as above, this log may be set a higher verbosity level to mark it less important

This is just for testing use, not actually needed

**Review by junotx — APPROVED** [review]

**Comment by benjaminhuo** [comment]
https://github.com/kubesphere/kube-events/pull/37 @junotx we can release a new version 0.4.1 after this PR is merged

**Pull request merged**